ext/phar/phar_object.c

```diff
@@ -1500,7 +1500,7 @@ static int phar_build(zend_object_iterator *iter, void *puser) /* {{{ */
 			}
 			ZEND_FALLTHROUGH;
 		default:
-			zend_throw_exception_ex(spl_ce_UnexpectedValueException, 0, "Iterator %s returned an invalid value (must return a string)", ZSTR_VAL(ce->name));
+			zend_throw_exception_ex(spl_ce_UnexpectedValueException, 0, "Iterator %s returned an invalid value (must return a string, a stream, or an SplFileInfo object)", ZSTR_VAL(ce->name));
```

Comment:
This feels like it should be a TypeError? I'm really not a fan of having SPL exceptions being thrown in non-SPL extensions, and Phar is the biggest offender of this.

I know this is somewhat unrelated, but I think a base PharException class an specialized dedicated exceptions would be better for where it makes sense, but this one feels like it should just be a standard TypeError (which I'm also confused why it's not enforced by a return type already).

Reply:
This comes from iterator data, so how are you going to enforce a return type?
In any case, changing the spl exceptions should be done separately imo and is a BC break.

Reply:
blerg I hate ext/phar....